src/algorithms/reco/ParticleFlow.h

```diff
@@ -0,0 +1,168 @@
+#include <edm4eic/ClusterCollection.h>
+#include <edm4eic/ReconstructedParticleCollection.h>
+// for algorithm configuration
+#include "algorithms/interfaces/WithPodConfig.h"
+#include "ParticleFlowConfig.h"
+
+namespace eicrecon{
+
+  // --------------------------------------------------------------------------
+  //! Particle Flow
+  // --------------------------------------------------------------------------
+  /*! This algorithm takes a collection of tracks (and their projections)
+   *  and a pair of calorimeter collections, from an ECal and an HCal,
+   *  and returns a collection of "particle flow objects", reconstructed
+   *  particles constructed from either tracks or calorimeter clusters.
+   *
+   *  As there will likely need to be multiple PF algorithms at any given
+   *  time, e.g. different eta regions may need to use different algorithms,
+   *  the particular choice of algorithm is set by a user-configurable
+   *  flag. The `process()` call will then run whichever algorithm is
+   *  specified.
+   */
+  class ParticleFlow : public WithPodConfig<ParticleFlowConfig> {
+
+    public:
+
+      // ----------------------------------------------------------------------
+      //! Projection Bundle
+      // ----------------------------------------------------------------------
+      /*! An intermediate struct to make calculations easier. Holds track
+       *  projections grouped together and the running sum of energy.
+       */
+      struct ProjectionBundle {
+        float energy = 0.;
+        std::vector<edm4eic::TrackPoint> projections;
+      };
+
+      // ----------------------------------------------------------------------
+      //! Merged Cluster
+      // ----------------------------------------------------------------------
+      /*! An intermediate struct to make calculations easier. Holds info on
+       *  calo clusters merged so far.
+       */
+      struct MergedCluster {
+        bool done = false;
+        int32_t pdg = 0;
+        float mass = 0.;
+        float charge = 0.;
+        float energy = 0.;
+        edm4hep::Vector3f momentum = {0., 0., 0.};
+        edm4hep::Vector3f weighted_position = {0., 0., 0.};
+        std::vector<edm4eic::Cluster> clusters;
+      };
+
+      // aliases for input types
+      //   - FIXME switch over to tracks when Track EDM is ready
+      using TrkInput     = std::pair<const edm4eic::ReconstructedParticleCollection*, const edm4eic::TrackSegmentCollection*>;
+      using CaloInput    = std::pair<const edm4eic::ClusterCollection*, const edm4eic::ClusterCollection*>;
+      using CaloIDs      = std::pair<uint32_t, uint32_t>;
+      using VecCaloInput = std::vector<CaloInput>;
+      using VecCaloIDs   = std::vector<CaloIDs>;
+
+      // aliases for internal types
+      using ClustMap      = std::map<edm4eic::Cluster, bool>;
+      using TrackMap      = std::map<edm4eic::ReconstructedParticle, bool>;
+      using ProjectMap    = std::map<edm4eic::TrackSegment, bool>;
```

Comment:
A map from thing to bool is not useful. A map is already that. If something is in the map, it is true; if it isn't, then it is false. Benefit of not defining a bool as second: you don't need to go an loop over everything first to set it to false. So, where `track_map[track] = false` use `track_map.erase(it)`.